numberOfProvinces.js

```diff
@@ -0,0 +1,88 @@
+/*
+n cities
+
+a - b - c
+
+i: n x n matrix
+o: total number of provinces
+
+directly or indirectly
+
+Input: isConnected = [[1,1,0],[1,1,0],[0,0,1]]
+Output: 2
+
+   0 1 2
+0 [1,1,0],
+1 [1,1,0],
+2 [0,0,1]
+
+*/
+
+const { Queue } = require('./utils/queue');
+
+const CONNECTED = 1;
+
+const getNeighbors = (node, matrix) => {
+    const currentRow = matrix[node].length;
+    const neighbors = [];
+
+    for (let neighborNode = 0; neighborNode < currentRow; neighborNode++) {
+        const isSelf = neighborNode === node;
+        const isChildren = matrix[node][neighborNode] !== CONNECTED;
```

Comment:
We use the term children for directed edges. This problem has exclusively undirected edges.

Also, this is named incorrectly. It's named `isChildren` (this should be `isNeighbor`). But in fact, the condition `matrix[node][neighborNode] !== CONNECTED` checks that `neighborNode` is NOT a neighbor. To name this variable accurately, you'd want to say `isNotNeighbor`. But, as we've previously discussed, it's not good to name things negatively. Instead, you should do something like `const isNeighbor = matrix[node][neighborNode] === CONNECTED` and then your condition can look like `if (!isNeighbor) continue;`